app/components/booking/extend-booking-dialog.tsx

```diff
@@ -0,0 +1,126 @@
+import { useCallback, useEffect, useState } from "react";
+import { CalendarIcon } from "lucide-react";
+import { useZorm } from "react-zorm";
+import { z } from "zod";
+import useFetcherWithReset from "~/hooks/use-fetcher-with-reset";
+import { isFormProcessing } from "~/utils/form";
+import { tw } from "~/utils/tw";
+import Input from "../forms/input";
+import { Dialog, DialogPortal } from "../layout/dialog";
+import { Button } from "../shared/button";
+import When from "../when/when";
+
+type ExtendBookingDialogProps = {
+  className?: string;
+  currentEndDate: string;
+};
+
+export const ExtendBookingSchema = z.object({
+  endDate: z.coerce.date().refine((endDate) => {
+    const now = new Date();
+    return endDate > now;
+  }, "Please select a date in future."),
+});
+
+export default function ExtendBookingDialog({
+  className,
+  currentEndDate,
+}: ExtendBookingDialogProps) {
+  const [open, setOpen] = useState(false);
+  const fetcher = useFetcherWithReset<{
+    error?: { message: string };
+    success: boolean;
+  }>();
+
+  const zo = useZorm("ExtendBooking", ExtendBookingSchema);
+  const disabled = isFormProcessing(fetcher.state);
+
+  function handleOpen() {
+    setOpen(true);
+  }
+
+  const handleClose = useCallback(() => {
+    setOpen(false);
+    fetcher.reset();
+  }, [fetcher]);
+
+  useEffect(
+    function closeOnSuccess() {
+      if (fetcher?.data?.success) {
+        handleClose();
+      }
+    },
+    [fetcher?.data?.success, handleClose]
+  );
+
+  return (
+    <>
+      <Button
+        variant="link"
+        className="justify-start rounded px-2 py-1.5 text-sm font-medium text-gray-700 outline-none data-[disabled]:pointer-events-none data-[disabled]:opacity-50 hover:bg-slate-100 hover:text-gray-700"
+        width="full"
+        onClick={handleOpen}
+      >
+        Extend booking
+      </Button>
+
+      <DialogPortal>
+        <Dialog
+          className={tw("lg:max-w-[450px]", className)}
+          open={open}
+          onClose={handleClose}
+          title={
+            <div className="flex size-10 items-center justify-center rounded-full bg-primary-25">
+              <div className="flex size-8 items-center justify-center rounded-full bg-primary-50">
+                <CalendarIcon className="size-4 text-primary-500" />
+              </div>
+            </div>
+          }
+        >
+          <div className="px-6 pb-4">
+            <h3 className="mb-4">Extend booking date</h3>
+
```

Comment:
Add a paragraph with some extra text. "Change the end date of your booking to a date in the future". If the booking is currently overdue, we should also add a second sentence. "Your booking is overdue, so its status will get updated to ongoing".